adam/adam_processing_service.py

```diff
@@ -1,288 +1,50 @@
-
-        Returns:
-            ephemeris: Ephemeris from file as a Pandas DataFrame
-        """
-
-        ephemeris_text = self.get_result_raw_ephemeris(run_number, force_update)
-        ephemeris = stk.io.ephemeris_file_data_to_dataframe(ephemeris_text.splitlines())
-        return ephemeris
-
-    def __update_results(self, force_update):
-        if force_update or self._detailedOutputs is None:
-            results = self.get_results()
-            self._summary = json.loads(results['outputSummaryJson'])
-            self._detailedOutputs = json.loads(results['outputDetailsJson'])
+from adam import PropagationParams, OpmParams, Project
+from adam import ApsRestServiceResultsProcessor, MonteCarloResults
+from adam import AuthenticatingRestProxy, RestRequests
 
 
 class AdamProcessingService:
     """The ADAM service handling operations related to jobs."""
 
-    def __init__(self, rest):
+    def __init__(self, rest=AuthenticatingRestProxy(RestRequests())):
         self._rest = rest
 
     def __repr__(self):
         return "Adam Processing Service Class"
 
-    def get_jobs(self, project):
-        """Get the jobs within a certain workspace (project).
-
-        Args:
-            project (str): The workspace (project) id.
-
-        Returns:
-            list: a list of jobs within the workspace (project).
-        """
-        code, response = self._rest.get(f'/projects/{project}/jobs')
-        return response
-
-    def get_job_results(self, project, job_id):
-        """Get the job results for a specific job for a specific project.
-
-        Args:
-            project (str): The workspace (project) id.
-            job_id (str): The job id.
-
-        Returns:
-            result (ApsResults): a result object that can be used to query for data about the
-            submitted job
-        """
-        results_processor = ApsRestServiceResultsProcessor(self._rest, project)
-
-        return BatchPropagationResults(results_processor, job_id)
-
     def execute_batch_propagation(self,
                                   project,
                                   propagation_params: PropagationParams,
-                                  opm_params: OpmParams) -> BatchPropagationResults:
+                                  opm_params: OpmParams,
+                                  object_id=None,
+                                  user_defined_id=None) -> MonteCarloResults:
         """Create a new job to run a batch propagation.
 
         Args:
-            project (str): The workspace (project) id.
+            project (str | Project): The workspace (project) id or project object
             propagation_params (PropagationParams): Parameters for the propagation.
             opm_params (OpmParams): Parameters specific to the OPM.
 
         Returns:
-            BatchPropagationResults: a reference to batch propagation object.
+            MonteCarloResults: a reference to batch propagation object.
```

Comment:
I changed the type name from BatchPropagationResults to MonteCarloResults because it is very specific to the Monte Carlo analysis and not a generic result type. Once we add OD processing the output results JSON will be very different and need its own type. ApsResults is the generic processor that only handles the JSON strings.